wrangler-core/src/main/java/io/cdap/wrangler/parser/GrammarBasedParser.java

```diff
@@ -95,8 +98,8 @@ public List<Directive> parse()
       return result;
     } catch (CompileException e) {
       throw new DirectiveParseException(e.getMessage(), e.iterator(), e);
```

Comment:
Why is DirectiveParseException not re-thrown as recipe exception?

Reply:
DirectiveParseExceptions are thrown from several different places (for ex, from initialise methods of directives, grammar parse methods etc). In most cases, the index of the directive is not easily available. So, it's not re-thrown here to keep it consistent with the other uses